packages/communication-react/review/communication-react.api.md

```diff
@@ -1213,14 +1214,12 @@ export interface JumpToNewMessageButtonProps {
 // @public
 export const lightTheme: PartialTheme & CallingTheme;
 
-// @public
-export const LocalizationProvider: (props: LocalizationProviderProps) => JSX.Element;
-
-// @public
-export type LocalizationProviderProps = {
-    locale: ComponentLocale;
+// @public (undocumented)
+export const LocalizationProvider: (props: {
+    defaultLocale?: string | undefined;
+    locale: ComponentLocale | ((locale?: string | undefined) => Promise<ComponentLocale>);
```

Comment:
Can we take a step back and understand the use-case this addresses.

Is the intention to

1. load the locale asynchronously, but only once at the start, based on hint from browser environment
2. be able to load a different locale based on user action
3. Allow Contoso to pick a locale at *build time* and only include that locale in the bundle.

I'm guessing the answer is 1. 

This API can't handle 2 well, because the Composite implementation won't know when a new locale is available and the whole composite will have to be refreshed when that happens.

If the intention is 3, we don't need any support for dynamic loading of locales.

Reply:
This PR actually covers all three of those use-cases. I just want to document what the intended use-cases are.

Reply:
My question is similar to commenting here, I'm confused on the functional use of the async locale loader.

Wouldn't Contoso just load the locale in an async fashion and then update the providers locale?

e.g.
```jsx

const [local, setLocale] = useState<Locale>(defaultLocale);

// inside useEffect:
(async () => {
  // fetch/load locale in an async manner
  const newLocale = await fetchOrLoadLocale('fr');
  setLocale(newLocale);
}());

<LocalizationProvider locale={choseLocale}>
  <Composite />
<LocalizationProvider />

```